$NetBSD: patch-test_Makefile.am,v 1.1 2023/06/17 11:25:19 riastradh Exp $

Use native compiler for executable compiled and run at build-time.

--- test/Makefile.am.orig	2023-04-18 19:55:55.000000000 +0000
+++ test/Makefile.am
@@ -27,7 +27,10 @@ testdir=$(pkgdatadir)/test
 
 test_DATA=$(check_SCRIPTS) $(TABLES)
 
-noinst_PROGRAMS=math-tables
+math-tables$(BUILD_EXEEXT): math-tables.o
+	$(CC_FOR_BUILD) -o $@ $(CFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) math-tables.o -lm
+math-tables.o: math-tables.c
+	$(CC_FOR_BUILD) -o $@ $(CPPFLAGS_FOR_BUILD) $(CFLAGS_FOR_BUILD) -c math-tables.c
 
 CLEANFILES=$(TABLES)
 
@@ -39,5 +42,5 @@ EXTRA_DIST=$(check_SCRIPTS)
 
 math.5c: $(TABLES)
 
-$(TABLES): math-tables
-	./math-tables > $(TABLES)
+$(TABLES): math-tables$(BUILD_EXEEXT)
+	./math-tables$(BUILD_EXEEXT) > $(TABLES)
